app/models/prison.rb

```diff
@@ -36,6 +36,48 @@ def unfiltered_offenders
     OffenderEnumerator.new(@code)
   end
 
+  def allocations
+    @allocations ||= Allocation.active(offenders.map(&:offender_no), @code)
+  end
+
+  Summary = Struct.new :allocated, :unallocated, :new_arrivals, :missing_info, keyword_init: true
+
+  def summary
+    @summary ||= begin
+      summary = if womens?
+                  offenders.group_by do |offender|
+                    if allocations.detect { |a| a.nomis_offender_id == offender.offender_no }
+                      :allocated
+                    elsif offender.has_case_information? && offender.complexity_level.present?
+                      :unallocated
+                    elsif offender.prison_arrival_date.to_date == Time.zone.today
+                      :new_arrival
+                    else
+                      :missing_info
+                    end
+                  end
+                else
+                  offenders.group_by do |offender|
+                    if offender.has_case_information?
+                      if allocations.detect { |a| a.nomis_offender_id == offender.offender_no }
+                        :allocated
+                      else
+                        :unallocated
+                      end
+                    elsif offender.prison_arrival_date.to_date == Time.zone.today
+                      :new_arrival
+                    else
+                      :missing_info
+                    end
+                  end
+                end
```

Comment:
Any chance we can DRY this up? It looks like the existence of a Complexity Level is the only difference between these two.

Or are you deliberately keeping these separate?